tests/graph/triangle/test_triangle_graph.py

```diff
@@ -0,0 +1,161 @@
+"""Tests for the TriangleGraph object."""
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+from landlab.graph.triangle.dual_triangle import DualTriGraph
+from landlab.graph.triangle.triangle_mesh import TriangleMesh
+
+if not TriangleMesh.validate_triangle():
+    pytestmark = pytest.mark.skip(reason="triangle is not installed")
+
+
+def test_graph_init():
+    graph = DualTriGraph(([0, 0, 10, 10], [0, 10, 10, 0]), triangle_opts="pqa1Djevz")
+
+    assert graph.number_of_nodes == 89
+    assert len(graph.x_of_node) == graph.number_of_nodes
+    assert len(graph.y_of_node) == graph.number_of_nodes
+    assert graph.number_of_links == 232
+    assert graph.number_of_patches == 144
+    assert graph.number_of_corners == graph.number_of_patches
+    assert graph.number_of_faces == 200
+    assert graph.number_of_cells == 57
+
+    assert np.all(graph.area_of_cell > 0.0)
+    assert np.all(graph.area_of_patch > 0.0)
+    assert np.all(graph.length_of_link > 0.0)
+    assert np.all(graph.length_of_face > 0.0)
+
+    assert_array_equal(
+        graph.nodes_at_patch[:3], [[41, 43, 15], [28, 31, 23], [51, 50, 4]]
+    )
+
+    assert_array_equal(graph.nodes_at_link[:3], [[41, 43], [43, 15], [15, 41]])
+
+    assert_array_equal(graph.nodes_at_face[:3], [[41, 43], [43, 15], [15, 41]])
+
+    assert_array_equal(graph.corners_at_face[:3], [[0, 66], [0, 65], [0, 6]])
+
+    assert_array_equal(
+        graph.corners_at_cell[:3],
+        [
+            [70, 79, 2, 77, 71, 68, 62, -1, -1],
+            [23, 65, 63, 27, 22, 28, -1, -1, -1],
+            [88, 115, 136, 101, 138, 89, 95, -1, -1],
+        ],
+    )
+
+    # assert_array_equal(graph.n_corners_at_cell[:3], [9, 9, 9])
```

Comment:
``DualTriGraph`` doesn't have a ``n_corners_at_cell`` (I don't think any of the graphs do). Do we need it? It looks like what was there before wasn't working correctly, anyway (the tested cells should have 7, 6, and 7 corners, not all 9).